lib/routes/users/routes.js

```diff
@@ -59,5 +75,129 @@ module.exports = [
                 }
             }
         }
+    },
+    {
+        method: 'POST',
+        path: '/register',
+        options: {
+            description: 'register new user.',
+            notes: 'POST users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    firstName: stringSchema,
+                    lastName: stringSchema,
+                    mobileNo: stringSchema,
+                    email: emailSchema,
+                    gender: Joi.string().valid('male', 'female', 'other')
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                let user = await registerUser(request.payload);
+                return h.response(user).code(200);
+            } catch (error) {
+                return h.response({ message: error.message }).code(400);
+            }
+        }
+    },
+    {
+        method: 'PUT',
+        path: '/profile/update',
+        options: {
+            description: 'update user profile.',
+            notes: 'PUT users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    id: idAllowedSchema,
+                    firstName: stringSchema,
+                    lastName: stringSchema,
+                    mobileNo: stringSchema,
+                    email: emailSchema,
+                    gender: Joi.string().valid('male', 'female', 'other')
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                await updateUser(request.payload);
+                return h
+                    .response({ message: 'User updated successfully.' })
+                    .code(200);
+            } catch (error) {
+                return h.response(error.message).code(400);
+            }
+        }
+    },
+    {
+        method: 'POST',
+        path: '/find/cab',
+        options: {
+            description: 'find nearby cabs',
+            notes: 'POST users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    pickupLocation: Joi.array().items(Joi.number())
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                let cab = transformDbArrayResponseToRawResponse(
+                    await findCab(request.payload.pickupLocation)
+                );
```

Comment:
Let's keep the logic of transform within the dao. It's a database pattern right.

Reply:
Done.